sdk/telemetry/test/conversion_test.go

```diff
@@ -0,0 +1,247 @@
+		ts.FromRaw("test=a")
+
+		s.SetParentSpanID(pcommon.SpanID([8]byte{0x1}))
+		// https://github.com/open-telemetry/opentelemetry-collector/issues/11267
+		// s.SetFlags(1)
+		s.SetName("span.a")
+		s.SetKind(ptrace.SpanKindClient)
+		s.SetStartTimestamp(pcommon.NewTimestampFromTime(y2k))
+		s.SetEndTimestamp(pcommon.NewTimestampFromTime(y2k.Add(time.Second)))
+		pAttrsA.CopyTo(s.Attributes())
+		s.SetDroppedAttributesCount(2)
+		prevent.CopyTo(s.Events().AppendEmpty())
+		s.SetDroppedEventsCount(3)
+		pLink.CopyTo(s.Links().AppendEmpty())
+		s.SetDroppedLinksCount(4)
+
+		stat := s.Status()
+		stat.SetMessage("okay")
+		stat.SetCode(ptrace.StatusCodeOk)
+
+		return s
+	}()
+	schema100 = "http://go.opentelemetry.io/schema/v1.0.0"
+
+	scope = &telemetry.Scope{
+		Name:         "go.opentelemetry.io/auto/sdk/telemetry/test",
+		Version:      "v0.0.1",
+		Attrs:        []telemetry.Attr{telemetry.String("department", "ops")},
+		DroppedAttrs: 1,
+	}
+	pScope = func() pcommon.InstrumentationScope {
+		s := pcommon.NewInstrumentationScope()
+		s.SetName("go.opentelemetry.io/auto/sdk/telemetry/test")
+		s.SetVersion("v0.0.1")
+		s.Attributes().PutStr("department", "ops")
+		s.SetDroppedAttributesCount(1)
+		return s
+	}()
+
+	scopeSpans = &telemetry.ScopeSpans{
+		Scope:     scope,
+		Spans:     []*telemetry.Span{spanA},
+		SchemaURL: schema100,
+	}
+	pScopeSpans = func() ptrace.ScopeSpans {
+		s := ptrace.NewScopeSpans()
+		pSpanA.CopyTo(s.Spans().AppendEmpty())
+		pScope.CopyTo(s.Scope())
+		s.SetSchemaUrl(schema100)
+		return s
+	}()
+
+	res = telemetry.Resource{
+		Attrs: []telemetry.Attr{
+			telemetry.String("host", "hal"),
+			telemetry.Int("id", 42),
+		},
+		DroppedAttrs: 100,
+	}
+	press = func() pcommon.Resource {
+		r := pcommon.NewResource()
+		r.Attributes().PutStr("host", "hal")
+		r.Attributes().PutInt("id", 42)
+		r.SetDroppedAttributesCount(100)
+		return r
+	}()
+
+	resSpans = &telemetry.ResourceSpans{
+		Resource:   res,
+		SchemaURL:  schema100,
+		ScopeSpans: []*telemetry.ScopeSpans{scopeSpans},
+	}
+	pResSpans = func() ptrace.ResourceSpans {
+		rs := ptrace.NewResourceSpans()
+		press.CopyTo(rs.Resource())
+		pScopeSpans.CopyTo(rs.ScopeSpans().AppendEmpty())
+		rs.SetSchemaUrl(schema100)
+		return rs
+	}()
+
+	traces = telemetry.Traces{
+		ResourceSpans: []*telemetry.ResourceSpans{
+			resSpans,
+		},
+	}
+	pTraces = func() ptrace.Traces {
+		traces := ptrace.NewTraces()
+		pResSpans.CopyTo(traces.ResourceSpans().AppendEmpty())
+		return traces
+	}()
+)
+
+func TestDecode(t *testing.T) {
+	var enc ptrace.JSONMarshaler
+	b, err := enc.MarshalTraces(pTraces)
+	require.NoError(t, err)
+
+	t.Log(string(b))
+
+	var got telemetry.Traces
+	dec := json.NewDecoder(bytes.NewReader(b))
+	require.NoError(t, dec.Decode(&got))
+
+	assert.Equal(t, traces, got)
+}
+
+func TestEncode(t *testing.T) {
+	var buf bytes.Buffer
+	enc := json.NewEncoder(&buf)
+	require.NoError(t, enc.Encode(traces))
+
+	data := buf.Bytes()
+	t.Log(string(data))
+
```

Comment:
nit: is this needed?

```suggestion
	t.Log(string(data))

```

Reply:
```suggestion
	t.Log(string(data)) // This helps when test fails to understand what how the data has been encoded.

```